Integration between current algorithms

PaulaViriato opened the pull request:
Addition of backtranslation algorithm, addition of deanonymization algorithm, integration between all algorithms, generation of a general dialog augmentation using all algorithms currently in the project.

PaulaViriato committed:
Integration between current algorithms

Addition of backtranslation algorithm, addition of deanonymization algorithm, integration between all algorithms, generation of a general dialog augmentation using all algorithms currently in the project.

jadermcs commented on backtranslation.py:
```diff
@@ -0,0 +1,216 @@
+    except: result = ""
+    return result
+
+def parallel_translation (original, rate, idioma, tradutor, currtags, traducoes):
```
In function declaration the Python style guide says to not put spaces between name and parenthesis, it should be in this case parallel_translation(...).

PaulaViriato replied:
Solved.

jadermcs commented on util.py:
```diff
@@ -0,0 +1,193 @@
+    new_turn["slot-values"] = turn["slot-values"].copy()
+    return new_turn
+
+def strTimeProp (start, end, format, prop):
```
python uses snake case, not camelcase, so the name should be str_time_prop(...)

PaulaViriato replied:
Solved.

jadermcs commented on mada.py:
```diff
@@ -1,79 +1,93 @@
-    return eda(sentence, ["[cpf]", "[placa]", "[valor]"])
-
-def bfs(pflow, intent_sample):
+def dfs (pflow, intent_sample, size, example, rate = 0.93, silence = False, mode = "test"):
```
This algorithm is now a Breadth-First Search, not a Depth anymore.

PaulaViriato replied:
Solved.

jadermcs reviewed (changes requested):
The code should improve style before merged.

PaulaViriato committed:
[#1] To fix requested changes: The code should improve style before merged.

- Function declaration without spacing;
- Snake case;
- Depth-First Search (dfs) to Breadth-First Search (bfs).

PaulaViriato committed:
Minor Fixes

- Acceptance of integer and string identifiers;
- Improved loop for cleaning and merging back-translated sentences;
- Added deanonymization for "categoria" entities;
- Treatment for null values of entities;
- Adjustment to commit 1240e2dda87bfe783934b918dc632bc4d5b53eab.

jadermcs committed:
Merge branch 'main' into main
